iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/Catalogs.java

```diff
@@ -284,4 +333,40 @@ private static String getCatalogType(Configuration conf, String catalogName) {
       }
     }
   }
+
+  public static class TableCache {
+    private static final Cache<String, Table> tableCache = Caffeine.newBuilder()
+        .expireAfterAccess(12, TimeUnit.HOURS).build();
```

Comment:
Shall we decrease this a bit? Perhaps 30 minutes would also suffice? Not sure what a "long" compilation time means for Hive usually. Decreasing it could give us some extra safety against cache pollution in case the hook-based eviction doesn't always go smoothly. Worst case in a very long compilation scenario, we might reload the table again eventually, but I'm assuming compilations that take 30+ minutes should be pretty rare?

Reply:
I kept the 12 hours threshold since we want to keep the cache entries until the execution is finished.